### DISCUSSION TITLE
First goal: Third party integration

### DISCUSSION THREAD

**Discussion by jgc31416** [bkkas/elhub-python-sdk#4] [Q&A]
One interesting point for a lot of people is to be able to use the third party integration against ElHub, there is some documentation here:
https://elhub.no/aktorer-og-markedsstruktur/testing/systemgodkjenning-system-vendor-trial/

We will  develop this library to be able to pass the system tests with as little development effort as possible, is it interesting for you?

**Comment by hakontro** [comment]
Hi, great initiative! Seems it currently consists of some early work done by you Jesus.

Are there any plans to continue the work on this? Would love to see some tutorials/docs to get us started, which would make it much easier for us to get up to speed so we can contribute later.

Håkon / sundaypower.no

**Reply by jgc31416** [reply]
Hi @hakontro thank you for the feedback,
Indeed the work is going on (I just pushed a new minor). We (Volte.no) have been able to pass the 3rd party tests in Elhub with the help of this library, there are some examples on how to pass some of the tests in the test folder /test/test_examples.

Perhaps we could create a document on the "Get Started" theme, what kind of info would you like to see there? There are many steps involved to start testing against ElHub, so I don't know where should the start point be :)

**Reply by hakontro** [reply]
Agree that the application process with ElHub should be out of scope for a Python SDK. I think having a guide that walks you through what's required to finish a single test-case from start to finish would be a great way to learn how to use this, and also verify that it's been set up correctly.

```
1. Clone SDK to local
2. Set up the env variables like so (suggest using something like https://pypi.org/project/python-dotenv/)
3. Start the test `AG622-01 Oppdatering av tredjepartstilgang (profil)` on Edielportalen.no
4. Run test script
5. Expected output should be like <whatever>
```

**Reply by jgc31416** [reply]
That makes sense, I will try to put a document with the steps to follow.
:+1: for the python-dotenv suggestion, it should make the package friendlier.

**Reply by hakontro** [reply]
Awesome! 

Another question: how long do requests take on your end? Seems they take several seconds to complete here, weird.. could just be ElHub?

**Reply by hakontro** [reply]
Haha, never mind 😭 
https://elhub.no/dokumentasjon-og-miljoer/miljoer/elhubs-testmiljo-for-aktorgodkjenning/

> 25.01.2023:
>BRS-NO-611 responderer tregt (opp mot 40 sekunder). Det samme gjelder også andre markedsprosesser og måleverdier, men her virker det å variere mer mellom når Elhub responderer som normalt og når det går tregt. I tillegg laster noen av rapportene i aktørportalen veldig tregt. Dette gjelder primært ‘Avansert søk’, ‘Forespørsel grunndata’ og ‘Grunndata sluttbruker’. Vi jobber med en løsning på disse problemene.

**Reply by jgc31416** [reply]
Well, on test I have seen consistently 20 secs, in production is usually a breeze, usually around 400ms
